src/asmcnc/apps/systemTools_app/screens/calibration/screen_stall_jig.py

```diff
@@ -0,0 +1,1660 @@
+
+    }
+
+    crash_distance = {
+
+        "X": 101,   # 381
+        "Y": 75,   # 65
+        "Z": -73    # -70
+
+    }
+
+    travel_to_stall_pos = {
+
+        "X": None,
+        "Y": None,
+        "Z": None
+
+    }
+
+
+    ## DIS/ENABLE MOTORS DEFINED IN CLASS INIT
+    ## AS PULLS FUNCTIONS FROM ROUTER_MACHINE
+
+    disable_motors = {}
+    enable_motors = {}
+
+    ## FLAGS FOR TEST EVENTS
+
+    setting_up_axis_for_test = False
+    expected_limit_found = False
+    threshold_reached = False
+    all_tests_completed = False
+    test_stopped = False
+    test_passed = False
+
+    ## CLOCK OBJECTS
+
+    poll_for_homing_completion_loop = None    
+    poll_for_ready_to_check_calibration = None
+    poll_for_ready_to_run_tests = None        
+    poll_for_going_to_start_pos = None        
+    poll_to_find_travel_from_start_pos = None 
+    poll_for_stall_position_found = None      
+    poll_for_threshold_detection = None       
+    poll_for_back_off_completion = None       
+    poll_to_relax_motors = None               
+    stadib_event = None                       
+    move_to_start_pos_event = None            
+    poll_to_prepare_to_find_stall_pos = None  
+    tell_user_ready_event = None              
+    poll_to_move_to_axis_start = None         
+    ensure_alarm_resumed_event = None         
+    threshold_detection_event = None          
+    hard_limit_found_event = None             
+    set_expected_limit_found_flag_event = None
+    poll_to_start_back_off = None             
+    drive_into_barrier_event = None           
+    move_all_axes_event = None                
+    poll_for_setting_up_axis_for_test = None  
+    get_alarm_info_event = None               
+    run_event = None                          
+    limits_acceleration_event = None          
+    post_move_all_axes_event = None           
+    restore_settings_event = None             
+    poll_to_finish_procedure = None           
+    data_send_event = None                    
+    resume_from_alarm_event = None            
+    threshold_reached = None
+    poll_to_deenergize_motors = None
+    poll_to_energize_motors = None
+    poll_to_reenable_hard_limits_and_go_to_next_test = None
+
+    ## DATABASE OBJECTS
+
+    id_stage = ""
+    stall_test_events = []
+    stall_test_data_col_names = [
+
+            "ID Stage: ",
+            "Axis: ",
+            "Test Feed: ",
+            "Threshold: ",
+            "Reported Feed: ",
+            "Load at detection: ",
+            "Stall coordinate: "
+    ]
+
+    stage_id = 9
+    
+
+    ## STORE ALL THE GRID BUTTONS
+
+    grid_button_objects = {}
+
+    ## COLOURS: 
+
+    fail_orange = [245./255, 127./255, 23./255, 1]
+    pass_green = [0./255, 204./255, 51./255, 1]
+    bright_pass_green = [51./255, 255./255, 0./255, 1]
+    highlight_yellow = [1, 1, 0, 1]
+    stop_red = [1, 0, 0, 1]
+    easycut_blue = [25./255, 118./255, 210./255, 1]
+
+    VERBOSE = True # For debugging
```

Comment:
Should this be set to true?

Reply:
Good spot! Thanks @DennisKiselev :). I think I'll leave it in for now, just while it starts being used in production 😁